### PR TITLE
Restore build pipeline

### PR DESCRIPTION
**Pull request by RussKie**
Addendum to #8522

## Proposed changes

- Build VC++ projects
To run locally run `dotnet build .\scripts\native.proj`
- Run localisation. 
To run locs locally run `dotnet build /p:Localise=true` (`--no-restore` is optional to speed it up)
- Run tests.
To run tests locally run `dotnet test` or run from VS
- Publish a portable.zip.   Msi can't be published due to https://github.com/wixtoolset/issues/issues/5627
To publish locally run `dotnet publish` (`--no-build` is optional to speed it up)

### CODE REVIEW

**Review by mstv — APPROVED** [review]
👍 Works well for me.

As already mentioned, some tests are failing (unrelated to this PR).

`TranslationApp.exe` is run despite the unexpected message (".dll" vs. ".exe"):
`The application to execute does not exist: 'D:\Build\gitextensions3\artifacts\Debug\bin\GitExtensions\net5.0-windows\TranslationApp.dll'.`

Build messages are a mix of German and English. I prefer dev tool output in English.

**Comment by mstv** [comment]
Build artifacts must be adapted yet:
```
C:\projects\gitextensions\GitExtensions\Project.Publish.targets(28,5): error MSB3030: Could not copy the file "C:\projects\gitextensions\\Externals\EasyHook\EasyHookDll\Build\Release\x64\EasyHook64.dll" because it was not found. [C:\projects\gitextensions\GitExtensions\GitExtensions.csproj]
```

File: appveyor.yml [hunk 1]
```diff
@@ -103,7 +96,7 @@ on_failure:
         Push-AppveyorArtifact "$_"
     }
 - ps: |
-    Get-ChildItem -recurse artifacts\TestsResults\Release\*.trx | `
+    Get-ChildItem -recurse artifacts\Release\TestsResults\*.trx | `
```
Comment (gerhardol) [inline appveyor.yml:99]: Release hardcoded

File: appveyor.yml [hunk 2]
```diff
@@ -50,41 +50,34 @@ build_script:
...
-
+    if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
 
 # scripts to run after tests
 after_test:
 - ps: |
     Write-Host "Preparing build artifacts..."
-    # & .\cibuild.cmd -publish -logFileName publish.binlog
-    # if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
+    dotnet publish -c Release --no-build /bl:.\artifacts\log\publish.binlog
```
Comment (gerhardol) [inline appveyor.yml:75]: Release hardcoded. Separate variable?

**Review by gerhardol — COMMENTED** [review]
+1

File: appveyor.yml (hunk 2 above)
Comment (RussKie) [reply appveyor.yml:76]: CI scripts must only build Release configuration.
I'm not sure how to define variables in yml, feel free to push a change.

File: appveyor.yml (hunk 1 above)
Comment (RussKie) [reply appveyor.yml:92]: No, this is the correct path:
image

**Comment by RussKie** [comment]
image

**Pull request merged**